src/types/token.type.ts

```diff
@@ -1,11 +1,17 @@
-export type Token = {
+import { ERC20Instance } from '@rsksmart/rif-relay-contracts/types/truffle-contracts';
+
+type TokenAttributes = {
     name: string;
-    decimals?: number;
-    symbol?: string;
-    contractAddress: string;
+    decimals: number;
+    symbol: string;
+};
+
+export type ERC20Token = Partial<TokenAttributes> & {
+    instance: ERC20Instance;
 };
```

Comment:
why is necessary to have the attributes and the instance itself in a structure like this?

Reply:
To give more flexibility for the user and not being querying the blockchain each time that the attribute is needed.

Reply:
> To give more flexibility for the user and not being querying the blockchain each time that the attribute is needed.

+1
@jurajpiar what if the users need some properties right now, but they still want to interact with the contract (i.e.: by calling a method?
In this way the user can decide what to do with it (or ignore it at all)